Client/src/Controllers/ResultPanelController.tsx

```diff
@@ -302,6 +305,12 @@ const mergeProps = (
   ]
 });
 
+function resultTypeHasChanged(prevResultType: ResultType, nextResultType: ResultType) {
+  return prevResultType.type === 'step' && nextResultType.type === 'step'
+    ? prevResultType.step.id !== nextResultType.step.id
+    : !isEqual(prevResultType, nextResultType);
+}
```

Comment:
Did you test this change with updating a step? I believe only checking the id will result the analysis not being reloaded. Perhaps we should compare `searchName` and `searchConfig`?

Reply:
D’oh - good catch! I just confirmed that invoking `loadTabs` is necessary for a revised step’s analyses to be reloaded.

I also confirmed that comparing `searchName` and `searchConfig` will indeed handle the case where we’re revising a step, and will also eliminate the redundant service call when adding a step.

Note to our future selves: we've also [considered a similar optimization in the Strategies Redux](https://github.com/VEuPathDB/WDKClient/blob/7af4b2b32e1747fce407913fb522827466e06e3d/Client/src/StoreModules/StrategyStoreModule.ts#L144), which might simplify this and similar comparisons of `resultType` in future client work.